third_party/ienn/src/ie_compilation.cpp

```diff
@@ -215,39 +224,60 @@ int32_t Compilation::AddConstant(uint32_t index,
         rank == 1
             ? Layout::C
             : rank == 2 ? Layout::HW : rank == 3 ? Layout::CHW : Layout::NCHW;
-    // GNA also only support FP32 representing with PREFER_ULTRA_LOW_POWER.
-    bool fp32_precision = preference_ != PREFER_LOW_POWER;
     Blob::Ptr blob;
-    if (fp32_precision) {
-      // GNA only accepts FP32 precision, cpu/gpu use FP32 currently.
-      blob = make_shared_blob<float>({Precision::FP32, dims, layout});
-    } else {
-      // MYRIAD only accepts FP16 precision.
-      blob = make_shared_blob<int16_t>({Precision::FP16, dims, layout});
-    }
-    blob->allocate();
-    const float* src =
-        reinterpret_cast<const float*>(model_->values[index].buffer);
     std::shared_ptr<op::Constant> constant_node;
     SizeVector const_dims = specified_dims.size() == 0 ? dims : specified_dims;
-    if (fp32_precision) {
-      float* dst = blob->buffer().as<float*>();
-      result = Reorder<float>(dst, src, operand.dimensions);
+
+    // CPU quant model accept uint8 data for weight
+    if (type == data_t::TENSOR_QUANT8_ASYMM) {
+      const uint8_t* src =
+          reinterpret_cast<const uint8_t*>(model_->values[index].buffer);
+      blob = make_shared_blob<uint8_t>({Precision::U8, dims, layout});
+      blob->allocate();
+
+      uint8_t* dst = blob->buffer().as<uint8_t*>();
+      result = Reorder<uint8_t, uint8_t>(dst, src, operand.dimensions);
       if (result != error_t::NOT_ERROR) {
         return result;
-      }
-      // The layout of constant_node depends on the shape we set
-      // e.g. 4D shape will use default NCHW
-      constant_node = std::make_shared<op::Constant>(
-          element::f32, Shape(const_dims), blob->buffer().as<float*>());
+        }
+        // The layout of constant_node depends on the shape we set
+        // e.g. 4D shape will use default NCHW
+        constant_node = std::make_shared<op::Constant>(
```

Comment:
The type of input and output could be different, such as input_type = fp32 and output_type = fp16. The Transpose operation couldn't cover this situation.

Reply:
Yes, I mean the same type in your PR `uint8 => uint8`

Reply:
Why do we need to add Transpose operand instead of Reorder?

Reply:
1, we can keep origin design
2, we can refactor with using `Transpose + convert` operands instead of the Reorder function, i am not sure whether it can be optimized in native graph, but i think we should use native ops as much as possible.
@huningxin please take a look.